Emby.Server.Implementations/Library/LibraryManager.cs

```diff
@@ -2408,6 +2410,38 @@ public bool FillMissingEpisodeNumbersFromPath(Episode episode, bool forceRefresh
                 episodeInfo = new Naming.TV.EpisodeInfo();
             }
 
+            try
+            {
+                var libraryOptions = GetLibraryOptions(episode);
+                if (libraryOptions.EnableEmbeddedEpisodeInfos && episodeInfo.Container.ToLowerInvariant() == "mp4") {
+                    // Read from metadata
+                    IMediaEncoder mediaEncoder = _appHost.Resolve<IMediaEncoder>();
```

Comment:
Don't create another MediaEncoder, use the existing one.

Reply:
I'm new to the code of Jellyfin, where do I find the existing one?

Reply:
In `ApplicationHost`.
Change the ctor of LibraryManager to take an `IMediaEncoder`

Reply:
But _appHost.Resolve<IMediaEncoder>() returns the mediaEncoder from ApplicationHost. So this is already the existing one. Have I missed something obvious?

Reply:
You are right, I mixed it up with something else. But it's still not the right solution, `IMediaEncoder` should get added to `LibraryManager` on creation.
DI is currently a bit of a mess, but it shouldn't be too hard to add.

Reply:
I updated the PR